Src/AutoFixture/DataAnnotations/MinAndMaxLengthAttributeRelay.cs

```diff
@@ -31,18 +41,117 @@ public object Create(object request, ISpecimenContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var minLengthAttribute = TypeEnvy.GetAttribute<MinLengthAttribute>(request);
-            var maxLengthAttribute = TypeEnvy.GetAttribute<MaxLengthAttribute>(request);
+            var range = Range.TryGetFromAttributes(request);
+            
+            if (range == null)
+            {
+                return new NoSpecimen();
+            }
+
+            var memberType = memberTypeResolver.TryGetMemberType(request);
+
+            if (IsString(memberType))
+            {
+                return ResolveString(context, range);
+            }
 
-            if (maxLengthAttribute == null && minLengthAttribute == null)
+            if (IsArray(memberType))
             {
+                return ResolveArray(context, memberType, range);
+            }
+
+            return new NoSpecimen();
+        }
+
+        private static bool IsString(Type memberType)
+        {
+            return memberType == typeof(string);
+        }
+
+        private static bool IsArray(Type memberType)
+        {
+            return memberType?.IsArray ?? false;
+        }
+        
+       private static object ResolveString(ISpecimenContext context, Range range)
+        {
+            return context.Resolve(new ConstrainedStringRequest(range.Min, range.Max));
+        }
+
+        private object ResolveArray(ISpecimenContext context, Type memberType, Range range)
+        {
+            var elementType = memberType.GetElementType();
+
+            var collectionCount = GetRandomNumberWithinRange(context, range);
+
+            if (collectionCount is NoSpecimen)
                 return new NoSpecimen();
+
+            var resultCollection = GetRandomCollection(context, elementType, collectionCount);
+
+            return ToArray((IEnumerable)resultCollection, elementType);
+        }
+
+        private static object GetRandomNumberWithinRange(ISpecimenContext context, Range range)
+        {
+            return context.Resolve(new RangedNumberRequest(typeof(int), range.Min, range.Max));
+        }
+
+        private static object GetRandomCollection(ISpecimenContext context, Type elementType, object collectionCount)
+        {
+            return context.Resolve(new FiniteSequenceRequest(elementType, (int)collectionCount));
+        }
+
+        private static object ToArray(IEnumerable elements, Type elementType)
```

Comment:
This function was copied from ArrayRelay. I would prefer to unify this logic somehow - @zvirja maybe some extension method for IEnumerable type? Or maybe you have some other suggestions?

Reply:
Let's keep that as is for now 😉 Usually we try to follow the rule of 3 and unify it only in that case. For now it's unclear what is the best way to do that. The idea with extension method would probably work as well, but let's postpone it for now.